docs/howto/customization/workflows.md

```diff
@@ -1,183 +1,559 @@
-    - name: GroundStateCalculation
+            - name: Output calculation
+              section: '../upload/archive/mainfile/dft.xml#/run/0/calculation/-1'
+    ```
+
+    When uploaded with `dft.xml` as before, this will generate a similar workflow graph, but with an extra input node.
+
+## Referencing Tasks in different uploads
+
+As already mentioned under "Considerations for archive path specification" above, your workflow YAML can reference entries that you have previously uploaded to NOMAD. In this case, you should replace the path prefix `../upload/archive/mainfile/<mainfile_name>` with `../uploads/<upload_id>/archive/<entry_id>`.
+
+??? Tip "Corresponding `dft.workflow.archive.yaml` from above example"
+
+    ```yaml
+    workflow2:
+      name: DFT SinglePoint
       inputs:
-        - name: input system
-          section: '../upload/raw/geom_opt.archive.yaml#/run/0/system/-1'
+        - name: Input system
+          section: '../upload/<upload_id>/archive/<entry_id>#/run/0/system/-1'
       outputs:
-        - name: ground state
-          section: '../upload/raw/ground_state.archive.yaml#/run/0/calculations/0'
+        - name: Output calculation
+          section: '../upload/<upload_id>/archive/<entry_id>#/run/0/calculation/-1'
+      tasks:
+        - m_def: nomad.datamodel.metainfo.workflow.TaskReference
+          task: '../upload/<upload_id>/archive/<entry_id>#/workflow2'
+          name: DFT
+          inputs:
+            - name: Input structure
+              section: '../upload/<upload_id>/archive/<entry_id>#/run/0/system/-1'
+          outputs:
+            - name: Output calculation
+              section: '../upload/<upload_id>/archive/<entry_id>#/run/0/calculation/-1'
+    ```
+
+## Nested workflows
+
+Nested, or hierarchical, workflows correspond to workflow graphs containing task nodes that themselves can be represented as a directed graph, i.e., a sub-workflow. The [General Workflow Schema](../../explanation/workflows.md#the-built-in-abstract-workflow-schema) allows for nested workflows through an inheritance relationship from the `Task` class to the `Workflow` class.
+
+### In multiple entries
+
+The most common way to construct a nested workflow is by creating separate entries for each (sub-)workflow. In this case, each sub-workflow archive will contain a populated `workflow2` section. Thus, to add a sub-workflow to your workflow YAML, the **best practice** is to directly link to this `workflow2` section, i.e., `task: <prefix>/<entry identifier>/workflow2`.
+
+!!! Warning "Important"
+    When `task` is linked to a `workflow2` section of a different upload, this sub-workflow task must be defined as a `TaskReference` by setting `m_def: nomad.datamodel.metainfo.workflow.TaskReference`. This is because `Workflow` instances can only contain `Task` instances and not reference them (see [General Workflow Schema](../../explanation/workflows.md#the-built-in-abstract-workflow-schema)).
```

Comment:
, otherwise this will use the default general baseclass `nomad.datamodel.metainfo.workflow.Task`

Reply:
I just find this statement `This is because `Workflow` instances can only contain `Task` instances and not reference them` a bit vague. you already mentioned the def to use if you want references, so you might as well say that the default class for workflow2.task is `nomad.datamodel.metainfo.workflow.Task`